envoy/server/factory_context.h

```diff
@@ -258,11 +258,23 @@ class FactoryContext : public virtual ListenerAccessLogFactoryContext {
    */
   virtual const Envoy::Config::TypedMetadata& listenerTypedMetadata() const PURE;
 
+  /**
+   * @return const bool if this listener is configured with bypass_overload_manager
+   *
+   */
+  virtual bool bypassOverloadManager() PURE;
+
   /**
    * @return OverloadManager& the overload manager for the server.
    */
   virtual OverloadManager& overloadManager() PURE;
 
+  /**
+   * @return NullOverloadManager& the dummy overload manager for the server for
+   * listeners that are bypassing a configured OverloadManager
+   */
+  virtual NullOverloadManager& nullOverloadManager() PURE;
```

Comment:
maybe interface should return an OverloadManager& instead of this subclass so that it's "opaque" whether we have one or the other. This would also allow us push the implementation outside of the interface file as currently done.